### PR TITLE
Add few more functions to RRT Star

### PR DESCRIPTION
**Pull request by americast**
@KgpKubs/path Added a few more functions - heuristic function to determine Grid id (hexagonal) and obstacle detector (between two given points). Also rectified the cost function.

### CODE REVIEW

File: src/RT_RRT_star.cpp [hunk 1]
```diff
@@ -54,25 +54,227 @@ namespace rrt {
...
+                  break;
+              }
+              if (RT_RRT<T>::line_path_obs(child,(RT_RRT<T>::tree[RT_RRT<T>::tree[j].second].first)))
+                here = cost(RT_RRT<T>::tree[RT_RRT<T>::tree[j].second].first,1,RT_RRT<T>::tree[j].second);
+              else
+                return std::pair<double,Utils::Point<T> > (std::numeric_limits<double>::infinity(),RT_RRT<T>::tree[RT_RRT<T>::tree[j].second].first);
+            }
+            else
+            {
+              if (RT_RRT<T>::line_path_obs(child,(RT_RRT<T>::tree[RT_RRT<T>::tree[j].second].first)))
```
Comment (kvmanohar22) [inline src/RT_RRT_star.cpp]: This seems to cause build error, did you mean to put this within the `for loop` ?

File: src/RT_RRT_star.cpp [hunk 2]
```diff
@@ -54,25 +54,227 @@ namespace rrt {
...
+                here = cost(RT_RRT<T>::tree[RT_RRT<T>::tree[k].second].first,1,RT_RRT<T>::tree[k].second);
+              else
+                return std::pair<double,Utils::Point<T> > (std::numeric_limits<double>::infinity(),RT_RRT<T>::tree[RT_RRT<T>::tree[j].second].first);
+            }
+
+            if (count)
+              return std::pair<double,Utils::Point<T> >  (dist(child,RT_RRT<T>::tree[j].second)+here.first,here.second);
```
Comment (kvmanohar22) [inline src/RT_RRT_star.cpp:164]: Same as above

File: src/RT_RRT_star.cpp [hunk 3]
```diff
@@ -54,25 +54,227 @@ namespace rrt {
...
+            else
+              return std::pair<double,Utils::Point<T> >  (dist(child,RT_RRT<T>::tree[j].second)+here.first,RT_RRT<T>::tree[RT_RRT<T>::tree[j].second].first);
```
Comment (kvmanohar22) [inline src/RT_RRT_star.cpp]: Same as above

**Review by kvmanohar22 — COMMENTED** [review]
Remove the binary file that is part of this PR as well.

File: src/RT_RRT_star.cpp (hunk 1 above)
Comment (americast) [reply src/RT_RRT_star.cpp]: Oops! It should have been `k`, not `j`. :stuck_out_tongue:

File: src/RT_RRT_star.cpp (hunk 2 above)
Comment (americast) [reply src/RT_RRT_star.cpp:164]: Same.

File: src/RT_RRT_star.cpp (hunk 3 above)
Comment (americast) [reply src/RT_RRT_star.cpp]: Same.

**Comment by americast** [comment]
@kvmanohar22 Removed the binary file! :slightly_smiling_face:

**Pull request merged**